### PR TITLE
Write error metadata

### PR DESCRIPTION
**Pull request by bjkeller**
Switching error capture from writing CSV files to adding errors to file metadata.

### CODE REVIEW

File: common/src/python/outputs/errors.py [hunk 1]
```diff
@@ -25,46 +20,16 @@ class JSONLocation(BaseModel):
...
-        Args:
-          location: the location object
-        Returns:
-          JSON string representation of the location object
-        """
-        if not location:
-            return None
-
-        return location.model_dump_json()
-
```
Comment (hewanadun) [inline common/src/python/outputs/errors.py:51]: don't we still need serialization for error_location?

**Review by hewanadun — APPROVED** [review]
Looks good. Please check the two comments.

File: common/src/python/outputs/errors.py [hunk 2]
```diff
@@ -25,46 +20,16 @@ class JSONLocation(BaseModel):
 class FileError(BaseModel):
     """Represents an error that might be found in file during a step in a
     pipeline."""
-    error_type: ErrorType
+    error_type: Literal['alert', 'error'] = Field(serialization_alias='type')
+    error_code: str = Field(serialization_alias='code')
     error_id: Optional[str] = None
```
[the remark this answers is not in the document]
Comment (bjkeller) [reply common/src/python/outputs/errors.py:25]: This was a change that Parker made on their side, which basically just flattens the structure

File: common/src/python/outputs/errors.py (hunk 1 above)
Comment (bjkeller) [reply common/src/python/outputs/errors.py:51]: No, since this will be stored in the metadata, it can be an object rather than a string

**Pull request merged**